File: heracles.client/src/types/product.ts
// src/types/product.ts
export type ProductImage = {
  id: number;
  productId: number;
  url: string;
  altText?: string | null;
  isPrimary: boolean;
  order: number;
};

export type ProductTag = {
  id: number;
  productId: number;
  tagId?: number | null;
  tagName: string;
};

export type Product = {
  id: number;
  name: string;
  price: number;
  description?: string | null;
  category?: string | null;
  stockQuantity: number;
  isActive: boolean;
  sku?: string | null;
  discountPercentage?: number | null;
  createdAt: string;
  updatedAt?: string | null;
  createdById: number;
  createdByEmail?: string | null;
  images?: ProductImage[];
  tags?: ProductTag[];
};

export type ProductCreateInput = {
  name: string;
  price: number;
  description?: string | null;
  category?: string | null;
  stockQuantity: number;
  isActive: boolean;
  sku?: string | null;
  discountPercentage?: number | null;
};

export type ProductUpdateInput = ProductCreateInput;

export type PaginatedProducts = {
  items: Product[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasPrevious: boolean;
  hasNext: boolean;
};

export type ProductFilters = {
  search: string;
  category: string;
  isActive: string;
  page: number;
  pageSize: number;
};

File: heracles.Server/Entities/Product.cs
namespace heracles.Server.Entities
{
    using System;

    public class Product
    {
        // Primary Key
        public int Id { get; set; }
        public Guid Uuid { get; set; } = Guid.NewGuid();
        // Champs de base
        public string Name { get; set; }
        public decimal Price { get; set; }

        // Champs supplémentaires
        public string? Description { get; set; }
        public string? Category { get; set; }
        public int StockQuantity { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public string? Sku { get; set; }
        public decimal? DiscountPercentage { get; set; }

        // ✅ CORRECTION : Changed from int to long
        public int CreatedById { get; set; }
        public User? CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        // NAVIGATION (NEW)
        public ICollection<ProductImage>? Images { get; set; }
        public ICollection<ProductTag>? Tags { get; set; }
    }
}